Add tips on Twitch/OBS streaming from @cg505 to staffdocs (#588)
ocf/ocfweb: ocfweb/docs/docs/staff/tips/twitch.md

[[!meta title="Using Twitch and OBS"]]

## Setting Up Your Stream

In your settings on Twitch, please enable VODs (video on demand) especially if you're working on OCF stuff.

> To do this, go to https://dashboard.twitch.tv/settings/channel and enable "Store past broadcasts".

Reviewer: IIRC this won't actually be a link on ocfweb. You need to use the `[]()` syntax to make it a link.

Author: I tested it with the original document, and ocfweb was still able to parse the link even without formatting it through Markdown. I've converted the links anyway though, but I found that interesting.

Reviewer: Oh neat, okay. Not sure where I got that impression then.

> When streaming, keep the stream manager open (ideally on another screen) to see chat: https://dashboard.twitch.tv/stream-manager.

Reviewer: ```suggestion
When streaming, keep the [stream manager](https://dashboard.twitch.tv/stream-manager) open (ideally on another screen) to see chat.
```
See previous comment about bare URLs not linking.

## Setting up OBS to stream to Twitch

> OBS is pretty straightforward to configure. You can install it as you would any other software (arch pkg: obs-studio).

Reviewer 2: link to their docs instead: https://obsproject.com/download

Reviewer 3: https://obsproject.com/wiki/ too

Reviewer: I said this on IRC, but it's worth noting that their docs want you to use a PPA on Ubuntu, which is not necessary. The page @dkess linked also only addresses Ubuntu. If we are going to link to them, we should use https://obsproject.com/wiki/Install-Instructions.

> Please record on OBS as well streaming! (Click both the "Start Streaming" and "Start Recording" buttons.)

Reviewer 3: while streaming

Reviewer: ```suggestion
Please record on OBS as well as streaming! (Click both the "Start Streaming" and "Start Recording" buttons.)
```
I think this is what I meant.

This will let us upload the streams to another source like YouTube. Twitch only saves VODs for a short period of time

> It's recommended to set up at least two scenes in OBS: one that shows your desktop and one that does not. This way, you can hide your screen if you need to work with passwords or otherwise sensitive stuff. Keep in mind that "it only showed for a split second" is not an excuse to not hide things. Twitch allows people to clip and rewatch anything frame-by-frame.

Reviewer 3: good info but not sure the tone is what we should be going for ("is not an excuse")

Reviewer: I don't want people to leak OCF secrets and be like "oh it's fine, I don't need to roll that password, it only showed for half a second!" I think that a lot of people have a mindset like that. That said, this can probably be reworded to be a little nicer and more clear.

Please stream with a mic so you can explain what you're doing and talk to chat. You can see the mic level in OBS. Ideally this should be hovering around the yellow range, but it should never hit the far right of the bar. If you're maxing out, decrease the mic volume. You can apply filters to the mic by clicking the gear icon.

> I (cooperc) would recommend adding Noise Supression (reduces noise) and Noise Gate (cuts off all sound below a certain volume level). For Noise Gate, to set the thresholds, be quiet and see where the mic level ends up. Set the close threshold there, and set the open threshold 10dB higher.

Reviewer 2: No need to say who the recommendation is coming from. I would change this to: "We recommend adding Noise Supression..."

> If you know what a compressor is, you can add one as well. (Not like gzip or mp3 you nerds. Audio compression means something entirely different.)

Reviewer 2: remove the snark or explain what you mean by audio compression

Reviewer: ```suggestion
If you know what a [compressor](https://en.wikipedia.org/wiki/Dynamic_range_compression) is, you can add one as well. (Not recommended if you aren't familiar.)
```
